### PR TITLE
JAMES-3136 Testing for failures & message projection

### PR DESCRIPTION
**Pull request by chibenwa**
- Source of truth
 - & retry strategy to limit inconsistencies

### CODE REVIEW

File: mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/mail/CassandraMessageIdMapperTest.java [hunk 1]
```diff
@@ -73,4 +81,128 @@ void findShouldReturnCorrectElementsWhenChunking() throws Exception {
...
+                // ignoring expected error
+            }
+
+            CassandraMessageIdToImapUidDAO imapUidDAO = new CassandraMessageIdToImapUidDAO(cassandra.getConf(), new CassandraMessageId.Factory());
+
+            SoftAssertions.assertSoftly(Throwing.consumer(softly -> {
+                softly.assertThat(sut.find(ImmutableList.of(message1.getMessageId()), MessageMapper.FetchType.Metadata))
+                    .hasSize(1);
+                softly.assertThat(imapUidDAO.retrieve((CassandraMessageId) message1.getMessageId(), Optional.empty()).collectList().block())
+                    .hasSize(1);
```
Comment (Arsnael) [inline mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/mail/CassandraMessageIdMapperTest.java:204]: Should we check it's empty in `MessageIdTable`?

Comment (chibenwa) [reply mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/mail/CassandraMessageIdMapperTest.java:204]: We made the insert fail so we know it is empty

File: mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMessageIdMapper.java [hunk 2]
```diff
@@ -127,9 +127,8 @@ public void save(MailboxMessage mailboxMessage) throws MailboxException {
         mailboxMapper.findMailboxById(mailboxId);
         ComposedMessageIdWithMetaData composedMessageIdWithMetaData = createMetadataFor(mailboxMessage);
         messageDAO.save(mailboxMessage)
-            .thenMany(Flux.merge(
-                imapUidDAO.insert(composedMessageIdWithMetaData),
-                messageIdDAO.insert(composedMessageIdWithMetaData)))
+            .thenEmpty(imapUidDAO.insert(composedMessageIdWithMetaData))
+            .thenEmpty(messageIdDAO.insert(composedMessageIdWithMetaData))
```
Comment (mbaechler) [inline mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMessageIdMapper.java]: can't we run this one concurrently with `updateIndexOnAdd`?

**Comment by chibenwa** [comment]
test this please

**Comment by chibenwa** [comment]
test this please

**Comment by chibenwa** [comment]
test this please

Build stalking for 16h, presented as aborted by JamesAdmin

**Comment by chibenwa** [comment]
test this please

Build stalking for 16h, presented as aborted by JamesAdmin

**Comment by Arsnael** [comment]
CI restart

test this please

**Comment by Arsnael** [comment]
Merged

**Pull request closed without merging**